### PR TITLE
Add refund calculation endpoint

### PR DESCRIPTION
**Pull request by builtbylane**

### CODE REVIEW

File: resources/refund.js [hunk 1]
```diff
@@ -22,4 +22,20 @@ function Refund(shopify) {
 
 assign(Refund.prototype, pick(baseChild, ['get', 'buildUrl']));
 
+
+/**
+ * Calculates a refund.
+ * https://help.shopify.com/api/reference/refund#calculate
```
Comment (lpinca) [inline resources/refund.js]: I'd remove this line as this is the only case where it is used.

File: resources/refund.js [hunk 2]
```diff
@@ -22,4 +22,20 @@ function Refund(shopify) {
...
+ *
+ * @param {Number} parentId Parent record ID
+ * @param {Object} [params] Query parameters
```
Comment (lpinca) [inline resources/refund.js]: This argument is not needed, please remove it.

File: resources/refund.js [hunk 3]
```diff
@@ -22,4 +22,20 @@ function Refund(shopify) {
...
+ * @return {Promise} Promise that resolves with the result
+ * @public
+ */
+Refund.prototype.calculate = function complete(orderId, params) {
```
Comment (lpinca) [inline resources/refund.js]: Ditto, please remove the `params` argument.

File: resources/refund.js [hunk 4]
```diff
@@ -22,4 +22,20 @@ function Refund(shopify) {
...
+  const url = this.buildUrl(orderId, '/calculate');
```
Comment (lpinca) [inline resources/refund.js]: Can you please remove the `/` from `'/calculate'`? It doesn't cause any harm but it's redundant.

File: resources/refund.js [hunk 5]
```diff
@@ -22,4 +22,20 @@ function Refund(shopify) {
...
+  return this.shopify.request(url, 'POST', undefined, params)
```
Comment (lpinca) [inline resources/refund.js]: Use `this.key` as third argument, remove `params`.

File: resources/refund.js [hunk 6]
```diff
@@ -22,4 +22,20 @@ function Refund(shopify) {
...
+    .then(body => body[this.key]);
```
Comment (lpinca) [inline resources/refund.js]: Remove this line, it's not needed if you add `this.key` as suggested in the previous comment.

**Comment by lpinca** [comment]
Thanks.
There is still some work to do:

1. Add tests.
2. Update the documentation.

**Comment by lpinca** [comment]
Merged as e6bb4f9596286ab7b8652268b7e70e6ca9731f53.
Thank you.

**Comment by builtbylane** [comment]
thanks for cleaning this up.

**Pull request closed without merging**